src/recipe-selector/recipe-selector.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Recipe} from '../runtime/recipe/recipe.js';
+import {Dictionary} from '../runtime/hot.js';
+
+export class RecipeSelector {
+    // Only recipes with “@trigger” annotations get included in the lookup table.
+    // Other recipes are silently ignored.
+    // If two triggers could match a request (one trigger is a subset of another) an exception is
```

Comment:
what about
```
@trigger
  A a
  B b
```
and
```
@trigger
  A a
  C c
```
?

Reply:
Exact matching, which is what I thought I'd implement as a first cut, is probably too restrictive. I'd like to allow overlapping triggers, preserve their order, and say that first match wins, but Piotr didn't like this. I'm not sure why. This will be a source of bikeshedding, to be sure. I'll change the comment and the doc, as our thinking develops.

Reply:
I actually did like first matching based on order - that's what I would like later with the UI allowing re-ordering of features. I was channelling Shane that we should aim for predictability in the system.